tensorboard/webapp/plugins/npmi/views/selected_annotations/selected_annotations_test.ts

```diff
@@ -0,0 +1,140 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+/**
+ * Unit tests for the Selected Annotations.
+ */
+import {TestBed} from '@angular/core/testing';
+import {By} from '@angular/platform-browser';
+
+import {Store, Action} from '@ngrx/store';
+import {provideMockStore, MockStore} from '@ngrx/store/testing';
+
+import {State} from '../../../../app_state';
+import {appStateFromNpmiState, createNpmiState} from '../../testing';
+import {SelectedAnnotationsContainer} from './selected_annotations_container';
+import {SelectedAnnotationsComponent} from './selected_annotations_component';
+import * as npmiActions from '../../actions';
+
+/** @typehack */ import * as _typeHackStore from '@ngrx/store';
+import {getPCExpanded, getSelectedAnnotations} from '../../store';
+
+describe('Npmi Selected Annotations', () => {
+  let store: MockStore<State>;
+  let dispatchedActions: Action[];
+  const css = {
+    TOOLBAR: By.css('.selected-annotations-toolbar'),
+    TITLE: By.css('.selected-annotations-title'),
+    BUTTONS: By.css('button'),
+    EXPAND_ICON: By.css('.expand-icon'),
+    EXPAND_LESS_ICON: By.css('.expand-less-icon'),
+  };
```

Comment:
I like how organized the selectors are up here!

Optional comments:

The title/icon checks seem like the most valuable part of this test.  I don't see the value in testing the existence of the `TOOLBAR` element.  It doesn't seem to carry any special rendering logic, and imo we should trust the Angular template syntax enough to drop it from test expectations.

Checking the `BUTTONS` is fine, perhaps relying on an explicit CSS class for the 'clear button' would be less fragile.  Sending a click to `buttons[0]` below might regress if we ever add more buttons / reorder them in the template.